### PR TITLE
Allow to use the script to concatenate with pseudo crawled data

### PR DESCRIPTION
**Pull request by thomasw21**
JSON file now needs to check for "is_catalogue" and change loading script depending on that.

### CODE REVIEW

File: aggregate_datasets.py [hunk 1]
```diff
@@ -40,13 +41,247 @@ def parse_args():
...
+            for row in zip(*[batch[name] for name in columns_to_collapse])
+        ]
+    }
+    return new_batch
+
+def collapse_meta(ds: Dataset, num_proc: int):
+    """{"text": str, "meta": str}"""
+    columns_to_keep = ["text"]
+    column_names_to_remove = [name for name in ds.column_names if name not in columns_to_keep]
+    return ds.map(collapse_meta_, batched=True, num_proc=num_proc, remove_columns=column_names_to_remove)
```
Comment (albertvillanova) [inline aggregate_datasets.py:264]: Could you please add inside "meta" a new field with the name of the source dataset?
```
{"text": "...", "meta": {..., "source_dataset": "lm_en_arxiv"}
```

You should agree with @yjernite which value to add for pseudo-crawl data (and if different for each seed/path?).

File: aggregate_datasets.py [hunk 2]
```diff
@@ -40,13 +41,247 @@ def parse_args():
...
+    if ds_ratio["is_catalogue"]:
+        ds = load_dataset(ds_name, use_auth_token=os.environ["HF_USER_ACCESS_TOKEN"], ignore_verifications=True)
+    else:
+        # We assume it comes from pseudo crawl.
+        # Pseudo crawl needs to be downloaded locally beforehand.
+        features = get_features()
+        dataset_path = Path(ds_name)
+        ds = load_dataset(str((dataset_path / "text__html").absolute()), data_files="**.jsonl.gz", features=features)
+        # collapse all meta data in "meta" column
+        ds = collapse_meta(ds, num_proc=1)
```
Comment (albertvillanova) [inline aggregate_datasets.py:279]: Once you add the field "source_dataset", maybe you could merge master branch into this branch and then move this line
```
# collapse all meta data in "meta" column
ds = collapse_meta(ds, num_proc=1)
```
to the `# Process meta:` section, again with an if for catalogue vs pseudo-crawl.

**Review by albertvillanova — CHANGES_REQUESTED** [review]
Some addition so that all meta are aligned.

File: aggregate_datasets.py (hunk 1 above)
Comment (thomasw21) [reply aggregate_datasets.py:264]: Right! Suggesting `f"pseudo-crawl--{seed_id}"`

**Review by albertvillanova — APPROVED** [review]
Thanks @thomasw21.

Let's see if the script finally works... :crossed_fingers:

**Comment by albertvillanova** [comment]
@thomasw21 I changed the way to read ratios once now they are JSONL instead of JSONL: https://github.com/bigscience-workshop/catalogue_data/commit/5eec922e0fe74ee52337fb90a817a1cb5abaf859

**Comment by thomasw21** [comment]
@albertvillanova can we revert that as it's in json format

**Comment by albertvillanova** [comment]
I realized afterwards. But I ketp the removal of `.iems()` from dset_ratios because now is a list, not a dict.

**Pull request merged**